components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -357,6 +362,35 @@ public Response subscriptionsSubscriptionIdGet(String subscriptionId, String ifN
         return null;
     }
 
+    @Override
+    public Response subscriptionsSubscriptionIdUsageGet(String subscriptionId, MessageContext messageContext) {
+
+        if (StringUtils.isBlank(subscriptionId)) {
+            String errorMessage = "Subscription ID cannot be empty or null when getting monetization usage.";
+            RestApiUtil.handleBadRequest(errorMessage, log);
+        }
+        try {
+            APIConsumer apiConsumer = RestApiUtil.getLoggedInUserConsumer();
+            Monetization monetizationImplementation = apiConsumer.getMonetizationImplClass();
+            Map<String, String> billingEngineUsageData = monetizationImplementation.
+                    getCurrentUsageForSubscription(subscriptionId, RestApiUtil.getLoggedInUserProvider());
+            if (MapUtils.isEmpty(billingEngineUsageData)) {
+                String errorMessage = "Billing engine usage data was not found for subscription ID : " + subscriptionId;
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+            APIMonetizationUsageDTO apiMonetizationUsageDTO = new APIMonetizationUsageDTO();
+            apiMonetizationUsageDTO.setProperties(billingEngineUsageData);
+            return Response.ok().entity(apiMonetizationUsageDTO).build();
+        } catch (APIManagementException e) {
+            String errorMessage = "Failed to retrieve billing engine usage data for subscription ID : " + subscriptionId;
+            RestApiUtil.handleInternalServerError(errorMessage, log);
+        } catch (MonetizationException e) {
```

Comment:
We must use `e` in the log in all error cases.

Reply:
Done in : https://github.com/wso2/carbon-apimgt/pull/6961/commits/0832832ce6d6c5e07bb85ed4d0846dafa034f88b

Reply:
Done in : https://github.com/wso2/carbon-apimgt/pull/6961/commits/0832832ce6d6c5e07bb85ed4d0846dafa034f88b